src/detect-transform-pcrexform.c

```diff
@@ -31,9 +31,11 @@
 #include "detect-transform-pcrexform.h"
 #include "detect-pcre.h"
 
+#define PCRE_ID_DATA_LEN 32
```

Comment:
this relates to the input regex string, right? What happens if that is > 32 chars? Why was 32 chosen, seems a bit arbitrary and needs a comment.

But can we not just retain the whole string?